pkg/controller/replicaset/replica_set.go

```diff
@@ -626,9 +626,9 @@ func (rsc *ReplicaSetController) manageReplicas(ctx context.Context, filteredPod
 		// expired even if other pods are deleted.
 		rsc.expectations.ExpectDeletions(rsKey, getPodKeys(podsToDelete))
 
-		errCh := make(chan error, diff)
+		errCh := make(chan error, len(podsToDelete))
 		var wg sync.WaitGroup
-		wg.Add(diff)
+		wg.Add(len(podsToDelete))
 		for _, pod := range podsToDelete {
```

Comment:
Sorry, could you tell me why we should use  len(podsToDelete) instead of diff？ As if we can not delete "diff" from this method, so it necessary for us to do this?

Reply:
In fact `diff` and l`en(podsToDelete)` are equal, this PR is to clarify that the number of wait groups is the same as the number of goroutines